### ISSUE TITLE
Getting "blocked as it could harm your computer"

### ISSUE THREAD

**Issue by ptallettms** [martinet101/ElevenClock#140]
Getting "blocked as it could harm your computer" when trying to download

**Comment by martinet101** [comment]
Yes, I know that, it is because the code is not signed and i can't solve that

**Comment by ptallettms** [comment]
No you don't need a certificate, Microsoft signs it for you.

Is it a desktop app? You will need to bundle it into an APPX.

Cheers,
Paul

**Comment by ptallettms** [comment]
I have a similar desktop app in the Store (SpaceExplorer). You probably need something like this:

MakeCertificate.bat (the common name (CN) is assigned by the Store via Visual Studio):

del cert.*
"C:\Program Files (x86)\Windows Kits\10\bin\x64\makecert.exe" -r -h 0 -n "CN=83000000-7000-4000-A000-C80000000000" -eku 1.3.6.1.5.5.7.3.3 -pe -sv cert.pvk cert.cer
"C:\Program Files (x86)\Windows Kits\10\bin\x64\pvk2pfx.exe" -pvk cert.pvk -spc cert.cer -pfx cert.pfx
Certutil -addStore TrustedPeople cert.cer

Then make a solution with Visual Studio and in your Post-Build command line add:

copy /Y $(TargetDir)\SpaceExplorer.exe $(ProjectDir)\CentennialPackage\
del d:\ptallett\SpaceExplorer\SpaceExplorer\CentennialPackage\DiskSpaceExplorer.appx
"c:\Program Files (x86)\Windows Kits\10\bin\x86\makeappx.exe" pack /d d:\ptallett\SpaceExplorer\SpaceExplorer\CentennialPackage\ /p d:\ptallett\SpaceExplorer\SpaceExplorer\CentennialPackage\DiskSpaceExplorer.appx
"C:\Program Files (x86)\Windows Kits\10\bin\x64\signtool.exe" sign -f d:\ptallett\spaceexplorer\spaceexplorer\cert.pfx -fd SHA256 -v d:\ptallett\SpaceExplorer\SpaceExplorer\CentennialPackage\DiskSpaceExplorer.appx

You should then be able to install your APPX and test. When you are happy, submit to the Microsoft Store. Once you have been approved to submit desktop apps, they will sign it for you.

I prefer the above direct approach, and you will need a manifest and some icons, but another approach is to use the VS Packager:

https://docs.microsoft.com/en-us/windows/msix/desktop/desktop-to-uwp-packaging-dot-net

The article is useful regardless of which way you go.

Cheers,
Paul

**Comment by martinet101** [comment]
Well, i was meaning (if it isn't difficult for you) to publish the app as one of your apps, and the built-in updater will install the next regular version. (Software like Icons8 desktop app do that (https://icons8.com/app/windows)).

But if it is difficult or you don't want to just don't feel forced to do it
